src/RawCodeCoverageData.php

```diff
@@ -0,0 +1,71 @@
+<?php declare(strict_types=1);
+/*
+ * This file is part of phpunit/php-code-coverage.
+ *
+ * (c) Sebastian Bergmann <sebastian@phpunit.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+namespace SebastianBergmann\CodeCoverage;
+
+/**
+ * Raw context-less code coverage data for SUT.
+ */
+final class RawCodeCoverageData
+{
+    /**
+     * Line coverage data.
+     *
+     * @see https://xdebug.org/docs/code_coverage for format
+     *
+     * @var array
+     */
+    private $lineData = [];
+
+    public function __construct(array $rawCoverage = [])
```

Comment:
I do not like the fact that the constructor has to guess the format of the array passed in. Can we please make the constructor private and add separate static methods, named constructors, for each supported format? For instance: `fromXdebugWithoutPathCoverage()`, `fromXdebugWithPathCoverage()`, etc.

Reply:
Sure thing

Reply:
Thanks!